src/workerd/api/streams/common.h

```diff
@@ -510,6 +510,10 @@ class ReadableStreamController {
     jsg::Lock& js, kj::Own<WritableStreamSink> sink, bool end) = 0;
 
   virtual kj::Own<ReadableStreamController> detach(jsg::Lock& js, bool ignoreDisturbed) = 0;
+
+  // Used by sockets to signal that the ReadableStream shouldn't allow reads due to pending
+  // closure.
+  virtual void setErrorOnReadsDueToPendingClosure(bool val) = 0;
```

Comment:
The name is a bit unfortunate. I'd probably go with something like `sendPendingClosure()` ... and is this ever set to `false`? If not, is the argument actually necessary?